x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/filters/filters.tsx

```diff
@@ -0,0 +1,282 @@
+      f.label === filter.label
+  ).length;
+
+const makeUniqueLabel = (filterArr: Filter[], filter: Filter) => {
+  let label = '';
+  let count = 0;
+  do {
+    count++;
+    label = `${filter.input.query} [${count}]`;
+  } while (filterArr.find((f) => f.label === label));
+  return label;
+};
+
+interface DraggableLocation {
+  droppableId: string;
+  index: number;
+}
+
+export interface FiltersIndexPatternColumn extends FieldBasedIndexPatternColumn {
+  operationType: 'filters';
+  params: {
+    filters: Filter[];
+  };
+}
+
+export const filtersOperation: OperationDefinition<FiltersIndexPatternColumn> = {
+  type: 'filters',
+  displayName: searchQueryLabel,
+  priority: 3, // Higher than any metric
+  getPossibleOperationForField: ({ type }) => {
+    if (type === 'document') {
+      return {
+        dataType: 'number',
+        isBucketed: true,
+        scale: 'ordinal',
+      };
+    }
+  },
+  isTransferable: () => false,
+
+  buildColumn({ suggestedPriority, field }) {
+    return {
+      label: searchQueryLabel,
+      dataType: field.type as DataType,
+      operationType: 'filters',
+      scale: 'ordinal',
+      suggestedPriority,
+      isBucketed: true,
+      sourceField: field.name,
+      params: {
+        filters: [],
+      },
+    };
+  },
+
+  toEsAggsConfig: (column, columnId) => ({
+    id: columnId,
+    enabled: true,
+    type: 'filters',
+    schema: 'segment',
+    params: {
+      filters: column.params.filters.length > 0 ? column.params.filters : [defaultSearchQuery],
+    },
+  }),
+  onFieldChange: (oldColumn, indexPattern, field) => {
+    return {
+      ...oldColumn,
+      label: field.displayName,
+      sourceField: field.name,
+    };
+  },
+  paramEditor: ({ state, setState, currentColumn, layerId }) => {
+    const indexPattern = state.indexPatterns[state.layers[layerId].indexPatternId];
+
+    const filters = currentColumn.params.filters;
+
+    const onDragEnd = ({
+      source,
+      destination,
+    }: {
+      source?: DraggableLocation;
+      destination?: DraggableLocation;
+    }) => {
+      if (source && destination) {
+        const items = euiDragDropReorder(filters, source.index, destination.index);
+        setFilters(items);
+      }
+    };
+
+    const setFilters = (newFilters: Filter[]) =>
+      setState(
+        updateColumnParam({
+          state,
+          layerId,
+          currentColumn,
+          paramName: 'filters',
+          value: newFilters,
+        })
+      );
+
+    return (
+      <EuiForm>
+        <EuiFormRow
+          label={i18n.translate('xpack.lens.indexPattern.filters.queries', {
+            defaultMessage: 'Queries',
+          })}
+        >
+          <div>
+            <EuiDragDropContext onDragEnd={onDragEnd}>
+              <EuiDroppable droppableId="FILTERS_DROPPABLE_AREA" spacing="s">
+                {filters?.map((filter: Filter, idx: number) => {
```

Comment:
Maybe we should assign a unique ID to each filter, like what esaggs does? It doesn't have to show to the user